client/errors.go

```diff
@@ -8,6 +8,11 @@ import (
 // ErrConnectionFailed is an error raised when the connection between the client and the server failed.
 var ErrConnectionFailed = errors.New("Cannot connect to the Docker daemon. Is the docker daemon running on this host?")
 
+// PermissionDenied returns an error with socketFile in the error message when a unix socket can not be opened.
+func PermissionDenied(socketFile string) error {
+	return fmt.Errorf("Cannot open %s. Permission Denied.", socketFile)
```

Comment:
Maybe something like "Cannot open %s. Permission Denied. If you are running the docker command as a non root user try using sudo before your command." ?